dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/instrumentation/decorator/HttpServerDecorator.java

```diff
@@ -132,7 +135,15 @@ public AgentSpanContext.Extracted extract(REQUEST_CARRIER carrier) {
     return extractContextAndGetSpanContext(carrier, getter);
   }
 
-  /** Deprecated. Use {@link #startSpan(String, Object, AgentSpanContext.Extracted)} instead. */
+  public Context extractContext(REQUEST_CARRIER carrier) {
+    AgentPropagation.ContextVisitor<REQUEST_CARRIER> getter = getter();
+    if (null == carrier || null == getter) {
+      return null;
+    }
+    return Propagators.defaultPropagator().extract(Context.root(), carrier, getter);
```

Comment:
nitpick: don't hesitate to use static import about the Context API. It was design for this use and would make it easier to read
```suggestion
    return defaultPropagator().extract(root(), carrier, getter);
```